documentation/source/benchmark.rst

```diff
@@ -0,0 +1,319 @@
+:class:`coreax.solvers.RandomSample`, and :class:`coreax.solvers.RPCholesky`.
+Each of these algorithms is evaluated across four different tests, providing a
+comparison of their performance and applicability to various datasets.
+
+Test 1: Benchmarking Coreset Algorithms on the MNIST Dataset
+------------------------------------------------------------
+
+The first test evaluates the performance of the coreset algorithms on the
+**MNIST dataset** using a simple neural network classifier. The process follows
+these steps:
+
+1. **Dataset**: The MNIST dataset consists of 60_000 training images and 10_000
+   test images. Each image is a 28x28 pixel grey-scale image of a handwritten digit.
+
+2. **Model**: A Multi-Layer Perceptron (MLP) neural network is used for
+   classification. The model consists of a single hidden layer with 64 nodes.
+   Images are flattened into vectors for input.
+
+3. **Dimensionality Reduction**: To speed up computation and reduce dimensionality, a
+   density preserving **UMAP** is applied to project the 28x28 images into 16 components
+   before applying any coreset algorithm.
+
+4. **Coreset Generation**: Coresets of various sizes are generated using the
+   different coreset algorithms. For :class:`coreax.solvers.KernelHerding` and
+   :class:`coreax.solvers.SteinThinning`, :class:`coreax.solvers.MapReduce` is employed
+   to handle large-scale data.
+
+5. **Training**: The model is trained using the selected coresets, and accuracy is
+   measured on the test set of 10_000 images.
+
+6. **Evaluation**: Due to randomness in the coreset algorithms and training process,
+   the experiment is repeated 5 times with different random seeds. The benchmark is run
+   on an **Amazon g4dn.12xlarge instance** with 4 NVIDIA T4 Tensor Core GPUs, 48 vCPUs,
+   and 192 GiB memory.
+
+**Results**:
+- Plots showing accuracy (with error bars) and time taken for each coreset size and
+algorithm.
+
+  .. image:: ../../examples/benchmarking_images/mnist_benchmark_accuracy.png
+     :alt: Benchmark Results for MNIST Coreset Algorithms
+
+  **Figure 1**: Accuracy of coreset algorithms on the MNIST dataset. Bar heights
+  represent the average accuracy. Error bars represent the min-max range for accuracy
+  for each coreset size across 5 runs.
+
+  .. image:: ../../examples/benchmarking_images/mnist_benchmark_time_taken.png
+     :alt: Time Taken Benchmark Results for MNIST Coreset Algorithms
+
+  **Figure 2**: Time taken to generate coreset for each coreset algorithm. Bar heights
+  represent the average time taken. Error bars represent the min-max range for each
+  coreset size across 5 runs.
+
+Test 2: Benchmarking Coreset Algorithms on a Synthetic Dataset
+--------------------------------------------------------------
+
+In this second test, we evaluate the performance of the coreset algorithms on a
+**synthetic dataset**. The dataset consists of 1_000 two-dimensional points,
```

Comment:
Rephrase something more like as "points in two-dimensional space", or "two-dimensional vectors"? As to me a point is typically zero-dimensional!